lib/module.gi

```diff
@@ -410,7 +410,7 @@ InstallMethod( DimensionOfVectors,
 # multiple immediate methods, several of which could apply and each changing
 # the type of the object. Doing so can be costly and thus should be
 # avoided.
-InstallOtherMethod(SetDimension,true,[IsObject and IsAttributeStoringRep,IsObject],
+InstallOtherMethod(SetDimension,true,[IsFreeLeftModule and IsAttributeStoringRep,IsObject],
```

Comment:
To justify this change, consider the declaration of `Dimension` in `lib/module.gd`, line 280:
```gap
DeclareAttribute( "Dimension", IsFreeLeftModule );
```
So in the GAP library, `Dimension` is only meant for `IsFreeLeftModule`, so it seems appropriate to restrict `SetDimension` to it.

That doesn't change that I still think FinInG should not use `Dimension` as it does, but it is more defensive and avoids the problem at hand.